### ISSUE TITLE
bem-drawer-menu

### ISSUE THREAD

**Issue by Bumerang47** [bem/bem-forum-content-ru#361]
Помогите пожалуйста подключить блок, ув. @Guria, (https://github.com/Guria/bem-drawer-menu. )
Вроде бы все элементы инициализировал, но браузер отвечает "SyntaxError: missing ) after argument list", ссылаясь на нало функции:
```
modules.define('menu', ['jquery', 'keyboard__codes'], function(provide, $, keyCodes, Menu) {
  provide(Menu.decl({ modName : 'fix', modVal : 'scroll' }
  {
    ...
```
Что делаю нет так.. ? Судя по ошибке какого-то аргумента не хватает.
Блоки в common.blocks добавил kg-menu и kg-glyph,
В bemjson блок меню создал по примеру из описания.

**Comment by apsavin** [comment]
Этот код - копипаст из редактора?
```js
provide(Menu.decl({ modName : 'fix', modVal : 'scroll' }, //<- вот здесь запятой не хватает
  {
```

**Comment by Guria** [comment]
@apsavin да. Я этот файл на гитхабе правил и не заметил ошибку. Исправлю сейчас.
@Bumerang47 не ожидал, что найдутся желающие использовать :) Для меня этот проект был как proof of concept.

**Comment by Bumerang47** [comment]
@apsavin, я из консоли браузера вставлял :)

@Guria. Хочу кнопку с меню для девайсов, Ваш вариант понравился, "почему бы и не опробовать" - подумал я =)
Спасибо.

**Comment by Bumerang47** [comment]
@Guria, покажите пожалуйста пример использования блока в проекте на bemjson. Без генерация. Как сделать что бы кнопка "Меню" изначально была на странице, без кнопки генерации. ._.

**Comment by Guria** [comment]
А сборка bh шаблонов у вас настроена?

**Comment by Bumerang47** [comment]
@Guria, хороший вопрос, пойду попробую выяснить )

**Comment by Guria** [comment]
То что он установлен, ещё не значит что он участвует в процессе сборки.
Какой сборщик, какой конфиг?

Кажется project-stub не приспособлен к сборке сразу 2мя шаблонизаторами. Да я даже не могу представить каким образом это можно сделать. Тут либо bemhtml, либо bh.

**Comment by Bumerang47** [comment]
@Guria, спасибо, теперь ястно. К сожалению таких тонкостей еще не освоил, посему изначально не догадался о конфликте. В руководстве по технологии bh, при установке о конфигурировании не видел информации, видать будет глубже. Пока уже bemhtml наворочен, в ближайшем будущем bh возьму за основу, для сравнения.

**Comment by Guria** [comment]
я сейчас ушёл на bemtree+bemhtml, потому что двухпроходность это добро. в bh 4.0 есть метод processBemjson и в теории с ним можно добиться того же самого, но вариантов настройки именно двухпроходности для bh я пока не встречал, да и xjst оказался не таким страшным как казался.
Единственный недостаток относительно bh, только не переопределяемость в runtime.

**Comment by tadatuta** [comment]
@Guria над этим недостатком сейчас идет работа, есть надежда, что проблема решаемая

**Comment by Guria** [comment]
Было бы интересно узнать как это. Пока сложно представить.

**Comment by tadatuta** [comment]
@vkz обещал написать подробности, когда все сойдется
